File: CodeBattleJava/src/main/java/ru/codebattle/client/SnakeBattleClient.java
package ru.codebattle.client;

import java.util.function.Function;
import ru.codebattle.client.api.GameBoard;
import ru.codebattle.client.api.SnakeAction;
import ru.codebattle.client.api.SnakeBattleBase;

import java.net.URISyntaxException;

public class SnakeBattleClient extends SnakeBattleBase {

    private Function<GameBoard, SnakeAction> callback;

    public SnakeBattleClient(String url) throws URISyntaxException {
        super(url);
    }

    public void run(Function<GameBoard, SnakeAction> callback) {
        connect();
        this.callback = callback;
    }

    @Override
    protected String doMove(GameBoard gameBoard) {
        clearScreen();
        gameBoard.printBoard();
        SnakeAction action = callback.apply(gameBoard);
        var command = action.toString();
        System.out.println(command);
        return command;
    }

    public void clearScreen() {
        System.out.print("\033[H\033[2J");
        System.out.flush();
    }

    public void initiateExit()
    {
        setShouldExit(true);
    }
}

File: CodeBattleNet/Client/SnakeBattleClient.cs
using System;
using SnakeBattle.Api;

namespace Client
{
    public class SnakeBattleClient : SnakeBattleBase
    {
        private Func<GameBoard, SnakeAction> _callback;

        public SnakeBattleClient(string serverAddress) : base(serverAddress)
        {
        }

        protected override string DoMove(GameBoard gameBoard)
        {
            //Just print current state (gameBoard) to console
            Console.Clear();
            //Console.SetCursorPosition(0, 0);
            gameBoard.PrintBoard();

            var action = _callback(gameBoard).ToString();
            Console.WriteLine(action);
            return action;
        }

        public void InitiateExit()
        {
            ShouldExit = true;
        }

        public void Run(Func<GameBoard, SnakeAction> callback)
        {
            Connect();
            this._callback = callback;
        }
    }
}
